[documentation] update project setup instructions in README (#1493)
sparkletown/sparkle: README.md

- https://create-react-app.dev/docs/adding-custom-environment-variables#adding-development-environment-variables-in-env

**Note:** Before you run the next steps, you will need to ensure you have access to the Firebase project you want to use. This access can be set up through the Firebase web UI. The following variables are required to be set up:

```bash
REACT_APP_PROJECT_ID=
REACT_APP_API_KEY=
REACT_APP_APP_ID=
REACT_APP_MEASUREMENT_ID=
```

> 

Reviewer: This seems like it might be better closer to/as a part of/directly after this line:

> In the main `sparkle` folder, copy `.env.example` to a new file `.env.local`, and fill it with the appropriate details.

As they are both relating to setting up the ENV vars.

npm start
```

> Once the server is started, your web browser will be opened at http://localhost:3000 (and then it'll be immediately redirected to https://sparklespaces.com/). Browser autorun can be opted out by adding a line to `.env.local`:

Reviewer: ```suggestion
Once the server is started, your web browser will be opened at http://localhost:3000 (and then it'll be immediately redirected to https://sparklespaces.com/). You can opt out of this by adding the following line to your `.env.local`:
```

### Firebase functions

**Note:** Before you run the following steps, you will need to ensure you have access to the Firebase project you want to use. This access can be set up through the Firebase web UI.

> 

Reviewer: We probably want to keep this note here in case people jump straight to this section, rather than following through the readme entirely from start to finish.

Author: How about on the original ordering, a link to and fro? Too complicated?

Author: The issue I had following the README linearly was it told me to run `npm start` before I had Firebase account or even the required variables set, so I was greeted with an error:
```
  Installations: Missing App configuration value: "projectId" (installations/missing-app-config-values).
  getMissingValueError
  C:/cloned/sparkletown/src/helpers/extract-app-config.ts:54
  ```

npm run firebase use staging

# Copy the runtime config locally

> npm run --silent firebase functions:config:get > ./functions/.runtimeconfig.json

Reviewer: Nice catch!

Seems we missed updating that when we added the `cd ..` part in https://github.com/sparkletown/sparkle/pull/1303

**Note**: Stripe is NOT REQUIRED unless you will be testing ticketing integration.

> First, you need to install the CLI. For Homebrew, use

Reviewer: ```suggestion
First, you need to install the [Stripe CLI](https://stripe.com/docs/stripe-cli).

If you use Homebrew, you can install it as follows:
```

```bash
brew install stripe/stripe-cli/stripe
```

> and for other environments, follow the instructions at [Stripe CLI](https://stripe.com/docs/stripe-cli#install)

Reviewer: ```suggestion
Otherwise, follow the installation instructions at https://stripe.com/docs/stripe-cli#install
```

> Make sure that you have a Stripe account with the right credentials. Then

Reviewer: ```suggestion
Make sure that you have a Stripe account setup and know the login credentials, then run the following commands (replacing `TODO-PROJECTID` with your actual Firebase project ID:
```

```bash
stripe login

> stripe listen --forward-to http://localhost:5001/co-reality-staging/us-central1/payment-webhooks

Reviewer: ```suggestion
stripe listen --forward-to http://localhost:5001/TODO-PROJECTID/us-central1/payment-webhooks
```
